tests/lib/rules/key-spacing.js

```diff
@@ -1058,5 +1058,217 @@ ruleTester.run("key-spacing", rule, {
+                beforeColon: false,
+                afterColon: true
+            },
+            multiLine: {
+                beforeColon: false,
+                afterColon: true
+            },
+            align: {
+                on: "colon",
+                beforeColon: true,
+                afterColon: true
+            }
+        }],
+        parserOptions: { ecmaVersion: 6, ecmaFeatures: { experimentalObjectRestSpread: true } },
+        errors: []
+    },
+    {
+        code: [
+            "({",
+            "    func:function() {",
+            "        var test = true;",
+            "    },",
+            "    longName: 1,",
+            "    small: 2,",
+            "    xs            : 3,",
+            "    func2    : function() {",
+            "        var test2 = true;",
+            "    },",
+            "    singleLine : 10",
+            "})"
+        ].join("\n"),
+        options: [{
+            singleLine: {
+                beforeColon: false,
+                afterColon: true
+            },
+            multiLine: {
+                beforeColon: false,
+                afterColon: true
+            },
+            align: {
+                on: "colon",
+                beforeColon: true,
+                afterColon: true,
+                mode: "strict"
+            }
+        }],
+        parserOptions: { ecmaVersion: 6, ecmaFeatures: { experimentalObjectRestSpread: true } },
+        errors: [
+            { message: "Missing space before value for key \'func\'.", line: 2, column: 10, type: "FunctionExpression" },
+            { message: "Missing space after key \'longName\'.", line: 5, column: 5, type: "Identifier" },
+            { message: "Missing space after key \'small\'.", line: 6, column: 5, type: "Identifier" },
+            { message: "Extra space after key \'xs\'.", line: 7, column: 5, type: "Identifier" },
+            { message: "Extra space after key \'singleLine\'.", line: 11, column: 5, type: "Identifier" }
+        ]
+    },
+    {
+        code: [
+            "({",
+            "    func: function() {",
+            "        var test = true;",
+            "    },",
+            "    longName: 1,",
+            "    small:    2,",
+            "    xs:       3,",
+            "    func2:    function() {",
+            "        var test2 = true;",
+            "    },",
+            "    singleLine: 10",
+            "})"
+        ].join("\n"),
+        options: [{
+            singleLine: {
+                beforeColon: false,
+                afterColon: true
+            },
+            multiLine: {
+                align: {
+                    on: "value",
+                    beforeColon: false,
+                    afterColon: true
+                },
+                beforeColon: false,
+                afterColon: true
+            }
+        }],
+        parserOptions: { ecmaVersion: 6, ecmaFeatures: { experimentalObjectRestSpread: true } },
+        errors: []
```

Comment:
Can you add two tests, one valid and one invalid, similar to this where `align` is an object inside `multiLine`, but where one or both of `beforeColon` or `afterColon` differ?